pkg/auth/token_provider.go

```diff
@@ -63,5 +73,15 @@ func (c *OIDCTokenProvider) GetJWT(serviceAccount types.NamespacedName, audience
 		return "", fmt.Errorf("could not request a token for %s: %w", serviceAccount, err)
 	}
 
+	// we need a duration until this token expires, use the expiry time - (now + 30s)
+	// this gives us a buffer so that it doesn't expire between when we retrieve it and when we use it
+	expiryTtl := tokenRequestResponse.Status.ExpirationTimestamp.Time.Sub(time.Now().Add(expirationBufferTime))
```

Comment:
What does Ttl stands for?

Reply:
This is the same terminology as the cache package uses, and it stands for "time to live". It is commonly used in scenarios like this where there is a duration of time for something to be valid for